proposals/function-pointers.md

```diff
@@ -0,0 +1,245 @@
+- Only a `funcptr` with a managed calling convention can be the target of such a conversion.
+
+This means developers can depend on overload resolution rules to work in conjunction with the 
+address-of operator:
+
+``` csharp
+unsafe class Util { 
+    public static void Log() { } 
+    public static void Log(string p1) { } 
+    public static void Log(int i) { };
+
+    funcptr void Action1();
+    funcptr void Action2();
+
+    void Use() {
+        Action1 a1 = &Log; // Log()
+        Action2 a2 = &Log; // Log(int i)
+
+        // Error: ambiguous conversion from method group Log to "void*"
+        void* v = &Log; 
+    }
+```
+
+The address-of operator will be implemented using the `ldftn` instruction.
+
+Restrictions of this feature:
+- Only applies to methods marked as `static`.
+- Local functions cannot be used in `&`. The implementation details of these methods are
+deliberately not specified by the language. This includes whether they are static vs. instance or
+exactly what signature they are emitted with.
+
+### Better function member
+The better function member specification will be changed to include the following line:
+
+> A `funcptr` is more specific than `void*`
+
+This means that it is possible to overload on `void*` and a `funcptr` and still sensibly use the address-of operator.
+
+## Open Issuess
+
+- Round tripping function pointer names, as well as parameter names, through metadata will require additional work. The
+function pointer type itself is natively supported by CLI but that does not include any names. This is not anticipated
+to be a big issue, just needs design work.
+- The address-of operator is limited to `static` methods in this proposal. It can be made to work with instance methods 
+but the behavior can be confusing to developers. The `this` type becomes an explicit first parameter on the `funcptr` 
+type. This means the behavior and usage would differ significantly from `delegate`. This extra confusion was the main 
+reason it was not included in the design.
+
+## Considerations
+
+### Don't require unsafe at declaration
+Instead of requiring `unsafe` at every use of a `funcptr`, only require it at the point where a method group is
+converted to a `funcptr`. This is where the core safety issues come into play (knowing that the containing assembly
+cannot be unloaded while the value is alive). Requiring `unsafe` on the other locations can be seen as excessive.
+
+This is how the design was originally intended. But the resulting language rules felt very awkward. It's impossible to
+hide the fact that this is a pointer value and it kept peeking through even without the `unsafe` keyword. For example 
+the conversion to `object` can't be allowed, it can't be a member of a `class`, etc ... The C# design is to require
+`unsafe` for all pointer uses and hence this design follows that.
+
+Developers will still be capable of preventing a _safe_ wrapper on top of `funcptr` values the same way that they do 
```

Comment:
What does it mean to "prevent a safe wrapper"? Did you mean "provide"?

Reply:
Yep I meant provide